BootloaderCorePkg/Tools/GenContainer.py

```diff
@@ -714,7 +723,7 @@ def create_container (args):
         hdr_entry[3] = args.auth
         container_list[0][0] = tuple(hdr_entry)
 
-    gen_container_bin (container_list, out_dir, comp_dir, key_dir, tool_dir)
+    gen_container_bin (container_list, out_dir, comp_dir, sign_scheme, key_dir, tool_dir)

```

Comment:
This might not the proper way to handle it.
Container might have multiple components and might have different signing scheme,  we should not use a top level parameter for this.  Instead,  this info needs to be put into the layout text to be used.

Reply:
Adding to layout we would need to enforce all container users to provide this input as part of layout.

#    Name ,  ImageFile      ,  CompAlg  ,  AuthType ,         Sign Schme  KeyFile                           ,  Size
# ====================================================================================================
  ( 'BOOT', 'IASIMAGE.bin'  , 'CLASSIC' , 'RSA2048_SHA2_256',  '**RSA_PSS**'  'TestSigningPrivateKey.pem'       , 0, 0),
  ( 'CMDL', 'cmdline.txt'   , 'Dummy'   , 'RSA2048_SHA2_256',  '**RSA_PSS**'  'TestSigningPrivateKey.pem'       , 0, 0),
  ( 'KRNL', 'vmlinuz'       , 'Lz4'    ,  'RSA2048_SHA2_256',  '**RSA_PSS**'  'TestSigningPrivateKey.pem'       , 0, 0),
  ( 'INRD', 'initrd'        , 'Lzma'    , 'SHA2_256',                      ''                                , 0, 0),

Alternatively could we add another auth type as,
RSA2048_SHA2_256_**PSS**
RSA3072_SHA2_384_**PSS** ?

Reply:
you alternative solution looks good to me.

Reply:
Thanks.  Container would support following auth types,
"NONE"               
"SHA2_256"            
"SHA2_384"             
"RSA2048_SHA2_256"     
"RSA3072_SHA2_384"    
"RSA**PSS**2048_SHA2_256"     
"RSA**PSS**3072_SHA2_384"